Makefile

```diff
@@ -25,14 +25,17 @@ build:
 	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -a -installsuffix cgo -ldflags '-s' -o $(BINDIR)/boot boot.go || exit 1
 
 docker-build:
+	# build the minio server
 	docker build -t minio mc
 	docker cp `docker run -d minio`:/go/bin/minio $(BINDIR)
+
+	# build the main image
 	docker build --rm -t ${IMAGE} rootfs
 	# These are both YAML specific
 	perl -pi -e "s|image: [a-z0-9.:]+\/deis\/${SHORT_NAME}:[0-9a-z-.]+|image: ${IMAGE}|g" ${RC}
 	perl -pi -e "s|release: [a-zA-Z0-9.+_-]+|release: ${VERSION}|g" ${RC}
 
-docker-push:
+docker-push: docker-build
```

Comment:
@mboersma I've discovered a reason for adding this dependency - if you want to `make docker-build && make docker-push` (a fairly common workflow), the image that `make docker-push` tries to push won't exist because image tags are time dependent (`$(shell date "+%Y%m%d%H%M%S")`).

I believe there was an idea thrown out somewhere to make image tags commit-sha dependent. I'm :+1: on that, but until we make a decision I'd like to leave as-is.